### DISCUSSION TITLE
Does Resizable make my React Children or Component client components

### DISCUSSION THREAD

**Discussion by NjengaFelix** [shadcn-ui/ui#2538] [General]
The Resizable component is a client component.

Does it mean anything I add inside resizable changes to a client component

**Comment by Alexjoshua14** [comment]
The Resizeable component does not result in everything inside being converted to a client component because it uses an interleaving pattern to essentially slot the components into place. Resizeable isn't effected by whether the components are server or client, it only needs to determine where they should be placed. The components would only be converted to client components if they are imported directly into a client component.

You can see more about this pattern on the [Next.js docs](https://nextjs.org/docs/app/building-your-application/rendering/composition-patterns#supported-pattern-passing-server-components-to-client-components-as-props) but essentially server components can be passed to client components as props. You can also import the Nextjs package 'server-only' into your server components to verify that the components remain server components. The package will throw a build-time error it is imported into a client component.

**Comment by NjengaFelix** [comment]
@Alexjoshua14 Thank  you so much for your detailed breakdown
